src/lib/crypto/payloads/StakePayload.ts

```diff
@@ -185,11 +187,11 @@ export class StakePayload extends TransactionPayloadMultiSig<StakeDepositParams>
         }
     }
 
-    protected async _estimateNetworkFees(provider: IProvider, priority = TransactionPriority.Medium): Promise<Nanowits> {
+    protected async _estimateNetworkFees(provider: IJsonRpcProvider, priority = TransactionPriority.Medium): Promise<bigint> {
         if (!this._priorities) {
             this._priorities = await provider.priorities()
         }
-        return Math.floor(
+        return BigInt(Math.floor(
```

Comment:
Why _estimateNetworkFees has to return a bigint? if the returning value is bigger than Number.MAX_SAFE_INTEGER then we have a problem with Math.floor

Reply:
because that's what callers to this method expect